src/hooks/useTitleChanges.js

```diff
@@ -0,0 +1,42 @@
+import { useMemo, useEffect, useCallback } from 'react'
+import { getAppFullName } from 'lib/utils.js'
+
+function useTitleChanges({ noteId, title, setTitle, serviceClient }) {
+  // Title change because of a local change from the editor
+  const onLocalTitleChange = useCallback(
+    serviceClient
+      ? e => {
+          const modifiedTitle = e.target.value
+          if (title != modifiedTitle) {
+            setTitle(modifiedTitle)
+            serviceClient.setTitle(noteId, modifiedTitle)
+          }
+        }
+      : () => {},
+    [noteId, title, setTitle, serviceClient]
+  )
+  // Title change because of a remote change from the stack
+  useEffect(
+    () => {
+      serviceClient &&
+        serviceClient.onTitleUpdated(noteId, modifiedTitle => {
+          if (title != modifiedTitle) {
+            setTitle(modifiedTitle)
+          }
```

Comment:
This hook adds an onTitleUpdated callback but does not deal with the cleanup of the callback, is it normal ? It is a bit surprising.

Reply:
in serviceClient, we only have one callback per event, so each new `onTitleUpdated` replace the previous one.